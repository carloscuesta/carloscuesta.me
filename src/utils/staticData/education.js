// @flow

export type Course = {
  dateFinish: ?string,
  dateStart: string,
  description: string,
  link: string,
  location: string,
  title: string
}

export type Education = Array<Course>

const educationHistory: Education = [
  {
    dateFinish: '06/05/2015',
    dateStart: '09/11/2013',
    description: 'A two years degree about building web applications. Using Front-End and Back-End technologies.',
    link: 'https://www.clot.fje.edu',
    location: 'Escola Jesu√Øtes El Clot',
    title: 'CFGS DAW'
  },
  {
    dateFinish: null,
    dateStart: '12/15/1995',
    description: 'The things I learned by myself became the vast majority of what I know today. #AlwaysLearning',
    link: '#',
    location: 'anywhere in the üåç',
    title: 'Self-taught'
  }
]

export default educationHistory
