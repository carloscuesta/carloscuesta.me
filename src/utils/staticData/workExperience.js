// @flow
export type JobPosition = {
  company: { link: string, logo: string, name: string },
  positions: Array<{
    dateFinish: ?string,
    dateStart: string,
    description: string,
    title: string
  }>
}

export type Experience = Array<JobPosition>

const workExperience: Experience = [
  {
    company: {
      link: 'https://ulabox.com',
      logo: 'https://res.cloudinary.com/carloscuesta/image/upload/v1593899513/about-me/ulabox.png',
      name: 'Ulabox'
    },
    positions: [
      {
        dateFinish: null,
        dateStart: '03/01/2018',
        title: 'Lead Front End Engineer',
        description: 'As a Lead Front End Engineer, one of my main tasks was to mentor coworkers of my team, contributing to their personal and professional growth. I worked very close to stakeholders to define, organise and breakdown new projects. One of the biggest challenges I faced with my team was to migrate the ulabox.com website to a new business model while refactoring the entire technology stack to Next.js and React. We had to integrate this pivot into the Ulabox application too. I\'ve worked with the logistics team to build Android applications with React-Native for our supply chain and dark stores.'
      },
      {
        dateFinish: '03/01/2018',
        dateStart: '01/09/2017',
        title: 'Front End Engineer',
        description: 'As a Front End Engineer I contributed to the development and release of the native Ulabox application. The application is built with React-Native and is available on iOS and Android. I worked on automating the process of shipping our applications to the stores using Fastlane and Travis-CI. Also I contributed to the maintenance of our website, that was built on top of Symfony using Twig, SCSS and JavaScript.'
      }
    ]
  },
  {
    company: {
      link: 'https://www.bemediatic.com',
      logo: 'https://res.cloudinary.com/carloscuesta/image/upload/v1593899512/about-me/mediatic.png',
      name: 'Medi√†tic'
    },
    positions: [
      {
        dateFinish: '12/23/2016',
        dateStart: '03/01/2015',
        title: 'Front End Developer',
        description: 'As a Front-End Developer I created a lot of different websites for our clients using Wordpress as a CMS and HTML, CSS and JavaScript to built the sites. I helped to improve the developer experience of the company by implementing tools to improve our software development process.'
      }
    ]
  },
  {
    company: {
      link: 'https://www.friendlyrentals.com',
      logo: 'https://res.cloudinary.com/carloscuesta/image/upload/v1593938160/about-me/friendlyrentals.png',
      name: 'Friendly Rentals'
    },
    positions: [
      {
        dateFinish: '02/11/2015',
        dateStart: '10/01/2014',
        title: 'Front End Developer Intern',
        description: 'As an intern on Friendly Rentals I helped on building pages of the website using Bootstrap, HTML, CSS and jQuery.'
      }
    ]
  }
]

export default workExperience
