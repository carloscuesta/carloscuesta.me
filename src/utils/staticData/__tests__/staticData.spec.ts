import education from '../education'
import workExperience from '../workExperience'

describe('staticData', () => {
  describe('education', () => {
    it('should match data', () => {
      expect(education).toMatchSnapshot()
    })
  })

  describe('workExperience', () => {
    it('should match data', () => {
      expect(workExperience).toMatchSnapshot()
    })
  })
})
