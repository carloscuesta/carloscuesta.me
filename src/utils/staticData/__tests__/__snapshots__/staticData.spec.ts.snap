// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`staticData education should match data 1`] = `
[
  {
    "dateFinish": "06/05/2015",
    "dateStart": "09/11/2013",
    "description": "A two years degree about building web applications. Using Front-End and Back-End technologies.",
    "link": "https://www.clot.fje.edu",
    "location": "Escola Jesu√Øtes El Clot",
    "title": "CFGS DAW",
  },
  {
    "dateFinish": null,
    "dateStart": "12/15/1995",
    "description": "The things I learned by myself became the vast majority of what I know today. #AlwaysLearning",
    "link": "#",
    "location": "anywhere in the üåç",
    "title": "Self-taught",
  },
]
`;

exports[`staticData workExperience should match data 1`] = `
[
  {
    "company": {
      "link": "https://n26.com",
      "logo": "https://res.cloudinary.com/carloscuesta/image/upload/v1617353828/about-me/n26.png",
      "name": "N26",
    },
    "positions": [
      {
        "dateFinish": null,
        "dateStart": "09/01/2022",
        "description": "",
        "title": "Lead Web Engineer",
      },
      {
        "dateFinish": "09/01/2022",
        "dateStart": "04/01/2021",
        "description": "",
        "title": "Senior Web Engineer",
      },
    ],
  },
  {
    "company": {
      "link": "https://ulabox.com",
      "logo": "https://res.cloudinary.com/carloscuesta/image/upload/v1593899513/about-me/ulabox.png",
      "name": "Ulabox",
    },
    "positions": [
      {
        "dateFinish": "03/26/2021",
        "dateStart": "03/01/2018",
        "description": "As a Lead Front End Engineer, one of my main tasks was to mentor coworkers of my team, contributing to their personal and professional growth. I worked very close to stakeholders to define, organise and breakdown new projects. One of the biggest challenges I faced with my team was to migrate the ulabox.com website to a new business model while refactoring the entire technology stack to Next.js and React. We had to integrate this pivot into the Ulabox application too. I've worked with the logistics team to build Android applications with React-Native for our supply chain and dark stores.",
        "title": "Lead Front End Engineer",
      },
      {
        "dateFinish": "03/01/2018",
        "dateStart": "01/09/2017",
        "description": "As a Front End Engineer I contributed to the development and release of the native Ulabox application. The application is built with React-Native and is available on iOS and Android. I worked on automating the process of shipping our applications to the stores using Fastlane and Travis-CI. Also I contributed to the maintenance of our website, that was built on top of Symfony using Twig, SCSS and JavaScript.",
        "title": "Front End Engineer",
      },
    ],
  },
  {
    "company": {
      "link": "https://www.bemediatic.com",
      "logo": "https://res.cloudinary.com/carloscuesta/image/upload/v1593899512/about-me/mediatic.png",
      "name": "Medi√†tic",
    },
    "positions": [
      {
        "dateFinish": "12/23/2016",
        "dateStart": "03/01/2015",
        "description": "As a Front-End Developer I created a lot of different websites for our clients using Wordpress as a CMS and HTML, CSS and JavaScript to built the sites. I helped to improve the developer experience of the company by implementing tools to improve our software development process.",
        "title": "Front End Developer",
      },
    ],
  },
  {
    "company": {
      "link": "https://www.friendlyrentals.com",
      "logo": "https://res.cloudinary.com/carloscuesta/image/upload/v1593938160/about-me/friendlyrentals.png",
      "name": "Friendly Rentals",
    },
    "positions": [
      {
        "dateFinish": "02/11/2015",
        "dateStart": "10/01/2014",
        "description": "As an intern on Friendly Rentals I helped on building pages of the website using Bootstrap, HTML, CSS and jQuery.",
        "title": "Front End Developer Intern",
      },
    ],
  },
]
`;
